src/main/cpp/org_zwave4j_Manager.cpp

```diff
@@ -2125,7 +2165,12 @@ JNIEXPORT void JNICALL Java_org_zwave4j_Manager_addWatcher
         env->NewGlobalRef(notificationWatcher),
         env->NewGlobalRef(context)
     );
-    notificationWatchers[std::pair<jobject, jobject>(notificationWatcher, context)] = pair;
+	
+	// This key is deallocated after this scope closes causing numerous issues.
```

Comment:
I'm quite poor at c++, but I thought that key is copied into a map as it is a value-type. Can you please give an example of issue?

Reply:
This was so long ago. I believe what was occurring was it was throwing an exception when removing the watcher because the key was garbage collected. By de-referencing the pointer to the actual structure into that table, it is never garbage collected until' "delete" or "free" is explicitly called on the pointer.